controllers/flinkcluster_converter.go

```diff
@@ -657,11 +657,29 @@ func getDesiredJob(observed *ObservedClusterState) *batchv1.Job {
 	}
 
 	if jobSpec.PythonFile != nil {
-		jobArgs = append(jobArgs, "--python", *jobSpec.PythonFile)
+		var pythonPath = *jobSpec.PythonFile
+		if strings.Contains(*jobSpec.PythonFile, "://") {
+			var parts = strings.Split(*jobSpec.PythonFile, "/")
+			pythonPath = path.Join("/opt/flink/job", parts[len(parts)-1])
+			envVars = append(envVars, corev1.EnvVar{
+				Name:  "FLINK_JOB_PYTHON_URI",
+				Value: *jobSpec.PythonFile,
+			})
+		}
+		jobArgs = append(jobArgs, "--python", pythonPath)
```

Comment:
this seems a bit duplicated. Can you move this into a separate fun?

Reply:
yes I have separated to a new function.
but the name of the function is unclear to know what it works exactly